### PR TITLE
ISO Unicast Add support for split configuration

### LINKED ISSUE #41596: Split connected ISO client and server by Kconfig
As a developer I want to support only one of the roles

Add a Kconfig option for each of the roles based on the central/peripheral Kconfigs

### PR DESCRIPTION
**Pull request by Thalley**
Add support for doing a ISO peripheral and a ISO central only configuration. 

fixes https://github.com/zephyrproject-rtos/zephyr/issues/41596

### CODE REVIEW

File: subsys/bluetooth/Kconfig.iso [hunk 1]
```diff
@@ -35,7 +51,6 @@ config BT_ISO_SYNC_RECEIVER
 	select BT_ISO_BROADCAST
 	select BT_OBSERVER
 	select BT_PER_ADV_SYNC
-	select EXPERIMENTAL
```
Comment (jhedberg) [inline subsys/bluetooth/Kconfig.iso:38]: You shouldn't remove this unless you also remove "[EXPERIMENTAL]" from the option description

Comment (cvinayak) [reply subsys/bluetooth/Kconfig.iso:38]: Experimental is now added to the parent BT_ISO instead, that should be ok, right?

Comment (jhedberg) [reply subsys/bluetooth/Kconfig.iso:38]: I thought "[EXPERIMENTAL]" in the option description and `select EXPERIMENTAL` always went hand-in-hand. @carlescufi can you confirm if this is the case or not?

Comment (Thalley) [reply subsys/bluetooth/Kconfig.iso:38]: I don't mind adding it. 

It's a good question: Should we also have `select EXPERIMENTAL` and `[EXPERIMENTAL]` for hidden Kconfigs, or just for those selectable?

Comment (Thalley) [reply subsys/bluetooth/Kconfig.iso]: Updated the Kconfigs such that we keep a 1:1 between `[EXPERIMENTAL]` and `select EXPERIMENTAL`, effectively removing it from the hidden Kconfigs and adding them to the selectable ones.

**Review by Casper-Bonde-Bose — DISMISSED** [review]
LGTM

**Comment by Thalley** [comment]
Rebased to solve merge conflict

**Comment by Thalley** [comment]
Fixed bad merge causing build error

**Comment by Thalley** [comment]
Rebased to fix CI

**Review by asbjornsabo — DISMISSED** [review]
LGTM

**Comment by Thalley** [comment]
Rebased to fix merge conflict

**Review by Casper-Bonde-Bose — DISMISSED** [review]
Reapprove after rebase

**Comment by Thalley** [comment]
Rebased to fix merge conflict

**Review by Casper-Bonde-Bose — APPROVED** [review]
Re-approve after rebase.

**Review by asbjornsabo — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@jhedberg could you please ACK since you are the assignee?

**Review by jhedberg — APPROVED** [review]

**Pull request merged**